### PR TITLE
Ready to Merge with master

### PR DESCRIPTION
**Pull request by 8Krzysztof8**
First part of task is ready to check

### CODE REVIEW

File: src/live/Cell.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	private Cell newCell;
+	private int chanceToSickness;
+	private int chanceToReproduce;
+	private int livetime;
+	private int age;
+	private boolean sick;
+
+	public Cell() {
+	}
+
```
Comment (Damian-B) [inline src/live/Cell.java]: remove additional empty line. Ctrl+Shift+F to invoke the Auto Formatter
It's important !!!

File: src/live/Cell.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+	public Cell(Cell newCell) {
+		this(newCell.chanceToSickness, newCell.chanceToReproduce, newCell.getLivetime(), newCell.isSick());
+	}
+
```
Comment (Damian-B) [inline src/live/Cell.java]: remove additional empty line Ctrl+Shift+F to invoke the Auto Formatter
It's important !!!

File: src/live/ClientCommunication.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package live;
+
+import java.util.Scanner;
+
+public class ClientCommunication {
+	static Scanner communication = new Scanner(System.in);
```
Comment (Damian-B) [inline src/live/ClientCommunication.java]: missing empty line Ctrl+Shift+F to invoke the Auto Formatter
It's important !!!

File: src/live/ClientCommunication.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+
```
Comment (Damian-B) [inline src/live/ClientCommunication.java]: remove additional empty line Ctrl+Shift+F to invoke the Auto Formatter
It's important !!!

File: src/live/ClientCommunication.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
+package live;
+
+import java.util.Scanner;
+
+public class ClientCommunication {
+	static Scanner communication = new Scanner(System.in);
+
+	
+	public static int readInt() {
+		int input = communication.nextInt();
```
Comment (Damian-B) [inline src/live/ClientCommunication.java]: Don't declare additional variables when You want to return them immediately !!!!!!11111111oneOneOne
return communication.nextInt();

!!!!!!!!!!

File: src/live/ClientCommunication.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+		Cell firstCell = new Cell();
+		System.out.println("Pass Cell chance to reproduce as procent");
+		firstCell.setChanceToReproduce(readInt());
+		System.out.println("You have just set Cell chance to reproduce to " + firstCell.getChanceToReproduce() + "\n Pass Cell chance to get sick as percent");
+		firstCell.setChanceToSickness(readInt());
+		System.out.println("You have just set chance to sickness to " + firstCell.getChanceToSickness()
+				+ "\n Pass the maximum age of cell");
+		firstCell.setLivetime(readInt());
+
```
Comment (Damian-B) [inline src/live/ClientCommunication.java]: remove additional empty line   Ctrl+Shift+F to invoke the Auto Formatter

File: src/live/ClientCommunication.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+		System.out.println("You have just set Cell chance to reproduce to " + firstCell.getChanceToReproduce() + "\n Pass Cell chance to get sick as percent");
+		firstCell.setChanceToSickness(readInt());
+		System.out.println("You have just set chance to sickness to " + firstCell.getChanceToSickness()
+				+ "\n Pass the maximum age of cell");
+		firstCell.setLivetime(readInt());
+		
+		
+		return firstCell;
+	}
+
```
Comment (Damian-B) [inline src/live/ClientCommunication.java]: remove additional emoty line Ctrl+Shift+F to invoke the Auto Formatter

File: src/live/Live.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Random;
+
+public class Live {
+
+	public static void main(String[] args) {
+
+		Logic obj = new Logic();
+		System.out.println("Beginning");
```
Comment (Damian-B) [inline src/live/Live.java]: You have ClientCommunication object it's good but why You use only to get some data ?
You can move every System.out to ClientCommunication and add there method "void display(message)". It allow You to change only one method in one place if You decide to change Your program from console to something different.

File: src/live/Live.java [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+		System.out.println("Pass number of cycles");
+		int numberOfCycles = ClientCommunication.communication.nextInt();
+		List<Cell> outcome = new LinkedList<Cell>();
+		outcome =firstCellProvider(outcome);
```
Comment (Damian-B) [inline src/live/Live.java]: code formating   =firstCell

File: src/live/Live.java [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+		
+		for (int i = 0; i < numberOfCycles; i++) {
+			outcome = obj.oneCycle(outcome);
+			for (Cell tempCell : outcome) {
+				tempCell.setAge(tempCell.getAge() + 1);
+				System.out.println(tempCell);
+			}
+			System.out.println("End of " + i + " cycle");
+		}
+
```
Comment (Damian-B) [inline src/live/Live.java]: remove line code formating

File: src/live/Live.java [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+	}
+
+	public static List<Cell> firstCellProvider(List<Cell> outcome) {
+		if (outcome.isEmpty()) {
+			Cell firstCell = ClientCommunication.CellPropertiesFiller();
+			outcome.add(firstCell);
+		}
+		return outcome;
+	}
+
```
Comment (Damian-B) [inline src/live/Live.java]: code formating....

File: src/live/Logic.java [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+			itr = undertaker(itr, tempCell);
+			itr = cellReproducer(itr, tempCell);
+			tempCell = sicknessMutator(tempCell);
+		}
+		return currentList;
+	}
+
+	public ListIterator<Cell> cellReproducer(ListIterator<Cell> itr, Cell tempCell) {
+		if (eventProbablity(tempCell.getChanceToReproduce())) {
+			nextCell = new Cell(tempCell);
```
Comment (Damian-B) [inline src/live/Logic.java]: Probably You can change 
                nextCell = new Cell(tempCell);
		itr.add(nextCell);
TO:
		itr.add(new Cell(tempCell));

File: src/live/Logic.java [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+	}
+
+	public ListIterator<Cell> undertaker(ListIterator<Cell> itr, Cell tempCell) {
+		if (tempCell.getAge() >= tempCell.getLivetime()) {
+			itr.remove();
+			System.out.println("Cell died");
+		}
+		return itr;
+	}
+
```
Comment (Damian-B) [inline src/live/Logic.java]: code formating

File: test/test/CellTest.java [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+	}
+
+	@Test
+	public void checkIfSkicknessModChangeCellToSick() {
+		Logic tempLogic = new Logic();
+		Cell tempCell = new Cell(100, 100, 10, false);
+		tempLogic.sicknessMutator(tempCell);
+		assertTrue(tempCell.isSick());
+	}
+	@Test
```
Comment (Damian-B) [inline test/test/CellTest.java]: add new line between } and Test

File: test/test/CellTest.java [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+		Cell tempCell2 = new Cell(100, 100, 10, false);
+		tempCell.setAge(11);
+		tempCell.setLivetime(5);;
+		tempCell2.setAge(12);
+		tempCell2.setLivetime(5);;
+		List<Cell> tempListOfCells = new LinkedList<Cell>();
+		ListIterator<Cell> itr = tempListOfCells.listIterator();
+		itr.add(tempCell);
+		itr.add(tempCell2);
+		System.out.println(itr.previous());  // This is just to track where itr is just diagnostic
```
Comment (Damian-B) [inline test/test/CellTest.java]: remove system.out your test are slower and you don;t know if you have access to System.out output

File: test/test/CellTest.java [hunk 16]
```diff
@@ -0,0 +1,80 @@
...
+		ListIterator<Cell> itr = tempListOfCells.listIterator();
+		itr.add(tempCell);
+		itr.add(tempCell2);
+		System.out.println(itr.previous());  // This is just to track where itr is just diagnostic 
+		System.out.println(itr.next());
+		tempLogic.undertaker(itr, tempCell);
+		System.out.println(itr.previous());
+		System.out.println(itr.next());
+		assertFalse(itr.hasNext());
+
```
Comment (Damian-B) [inline test/test/CellTest.java]: remove empty lines

File: src/live/Live.java (hunk 9 above)
Comment (8Krzysztof8) [reply src/live/Live.java]: ?? Nie kumam

**Pull request closed without merging**